types/validator.go

```diff
@@ -108,13 +108,38 @@ func (vc validatorCodec) Compare(o1 interface{}, o2 interface{}) int {
 
 // RandValidator returns a randomized validator, useful for testing.
 // UNSTABLE
-func RandValidator(randPower bool, minPower int64) (*Validator, *PrivValidatorFS) {
-	_, tempFilePath := cmn.Tempfile("priv_validator_")
-	privVal := GenPrivValidatorFS(tempFilePath)
+func RandValidator(randPower bool, minPower int64) (*Validator, Signer) {
+	privVal := GenSigner()
 	votePower := minPower
 	if randPower {
 		votePower += int64(cmn.RandUint32())
 	}
-	val := NewValidator(privVal.GetPubKey(), votePower)
+	val := NewValidator(privVal.PubKey(), votePower)
 	return val, privVal
 }
+
+type Signer interface {
```

Comment:
2 Signer interfaces. Wont this be confusing?

Reply:
yes this one is just for testing tho. problem of circular imports ...

Reply:
I'll change it to TestSigner